test/parse/test_entity_declaration.rb

```diff
@@ -23,6 +23,158 @@ def parse(internal_subset)
     end
 
     public
+
+    class TestGarbagedValue < self
```

Comment:
Could you use syntax part name such as `GeneralEntityDeclration` instead of `GarbgedValue` for test case name? We want to organize tests based of syntax part not invalid type to avoid missing case.

```suggestion
    class TestGeneralEntityDeclration < self
      # Cover all GEDecl cases in this instead of spreading multiple test cases
      def test_invalid_entity_value
      def test_invalid_external_id
```

Could you also add generic entity declaration definition (and related definitions in this test case) as a comment for easy to understand?
* https://www.w3.org/TR/2006/REC-xml11-20060816/#NT-GEDecl